staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -111,9 +120,11 @@ func SchemaDeclType(s *schema.Structural, isResourceRoot bool) *DeclType {
 				if s.ValueValidation != nil && s.ValueValidation.MaxProperties != nil {
 					maxProperties = zeroIfNegative(*s.ValueValidation.MaxProperties)
 				} else {
-					maxProperties = estimateMaxAdditionalPropertiesPerRequest(s.AdditionalProperties.Structural)
+					maxProperties = estimateMaxAdditionalPropertiesFromMinSize(propsType.MinSerializedSize)
 				}
-				return NewMapType(StringType, propsType, maxProperties)
+				mapType := NewMapType(StringType, propsType, maxProperties)
+				mapType.MinSerializedSize = estimateMinSizeForMap(s)
```

Comment:
estimateMinSizeForMap is iterating over `Properties`... do we need to do that for a map defined with `AdditionalProperties`? I thought this made them mutually exclusive:

https://github.com/kubernetes/kubernetes/blob/eb2f56c638b393fd589f4bda7b7b84d4d2ccfbdf/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L906-L920

which would make MinSerializedSize just 2, right?

Reply:
Whoops, that's true. I'll change that and remove `estimateMinSizeForMap`.